addons/mail/models/reset_template_mixin.py

```diff
@@ -0,0 +1,106 @@
+            # Process a single PO (or POT) entry
+            for row in reader:
+                if row.get('imd_name') in xml_ids:
+                    # copy Translation from Source to Destination object
+                    self.env['ir.translation']._set_ids(
+                        row['name'], 'model', lang, self._ids, row.get('value', ''), row['src'],
+                    )
+
+    def _override_translation_term(self, module_name, xml_ids):
+        for code, _ in self.env['res.lang'].get_installed():
+            lang_code = tools.get_iso_codes(code)
+            base_lang_code = lang_code.split('_')[0] if '_' in lang_code else lang_code
+            is_po_file_available = False
+            # In case of sub languages (e.g fr_BE), load the base language first, (e.g fr.po) and
+            # then load the main translation file (e.g fr_BE.po)
+
+            # Step 1: reset translation terms with base language file
+            base_trans_file = get_module_resource(module_name, 'i18n', base_lang_code + '.po')
+            if base_trans_file:
+                is_po_file_available = True
+                self._process_translation_data(base_trans_file, code, xml_ids)
+
+            # Step 2: reset translation file with main language file (can possibly override the
+            # terms coming from the base language)
+            trans_file = get_module_resource(module_name, 'i18n', lang_code + '.po')
+            if trans_file:
+                is_po_file_available = True
+                self._process_translation_data(trans_file, code, xml_ids)
+
+            # If no translation file available, unlink custom translated terms linked to template
+            if not is_po_file_available:
+                self.env['ir.translation'].search([
+                    ('name', 'like', f'{self._name},%'),
+                    ('res_id', '=', self.id),
+                    ('lang', '=', code),
+                    ('module', 'in', [module_name, False]),
+                ]).unlink()
+
+    def reset_template(self):
+        """Resets the Template with values given in source file. We ignore the case of
+        template being overridden in another modules because it is extremely less likely
+        to happen. This method also tries to reset the translation terms for the current
+        user lang (all langs are not supported due to costly file operation). """
+        for template in self.filtered(lambda t: t.template_fs and t.is_template_modified):
+            external_id = template.get_external_id().get(template.id)
+            module, xml_id = external_id.split('.')
+            fullpath = get_resource_path(*template.template_fs.split('/'))
+            if fullpath:
+                doc = etree.parse(fullpath)
+                for rec in doc.xpath("//record[@id='%s'] | //record[@id='%s']" % (external_id, xml_id)):
+                    obj = xml_import(template.env.cr, module, {}, mode='init', xml_filename=fullpath)
```

Comment:
Just to be sure, is it correctly checking write access on the record ? Code in xml_import is not that fluid so asking to be sure :sweat_smile:

Reply:
Who can trigger this action anyway? Isn't it a bit too permissive that any employee can reset a template?